integration/networking/port_mapping_linux_test.go

```diff
@@ -0,0 +1,206 @@
+
+	var (
+		err                      error
+		sgmt                     *testutils.L2Segment
+		victimHost, attackerHost testutils.L3Host
+		attackerNs               testutils.Netns
+	)
+
+	sgmt, err = testutils.NewL2Segment(t, "br-l2-segment", sgmtNw, testID, reusePrevious)
+	if err != nil {
+		return fmt.Errorf("failed to create L2Segment: %w", err)
+	}
+
+	// The current netns is used as this is where the container port is published.
+	victimNs := testutils.CurrentNetns(t)
+	victimHost = testutils.L3Host{
+		Ns:               victimNs,
+		HostIfaceName:    fmt.Sprintf("victim-%8x", testID),
+		BridgedIfaceName: "victim",
+	}
+	if err := sgmt.AddHost(&victimHost, reusePrevious); err != nil {
+		return fmt.Errorf("failed to create victim host: %w", err)
+	}
+
+	attackerNs = testutils.NewNamedNetns(t, fmt.Sprintf("attacker-%8x", testID), reusePrevious)
+	attackerHost = testutils.L3Host{
+		Ns:               attackerNs,
+		HostIfaceName:    "eth0",
+		BridgedIfaceName: "attacker",
+	}
+	if err := sgmt.AddHost(&attackerHost, reusePrevious); err != nil {
+		return fmt.Errorf("failed to create attacker host: %w", err)
+	}
+
+	t.Cleanup(func() {
+		if !manualDebug {
+			attackerNs.Destroy(t)
+			sgmt.Destroy(t)
+			return
+		}
+
+		fmt.Println("L2 segment is kept for manual debugging:")
+		fmt.Printf("\t* Bridge netns: %s\n", sgmt.BridgeNs)
+		fmt.Printf("\t* Attacker netns: %s\n", attackerNs)
+		fmt.Printf("\t* Victim MAC: %s\n", victimHost.MACAddr)
+		fmt.Printf("\t* Attacker MAC: %s\n", attackerHost.MACAddr)
+		fmt.Printf("\t* Victim IP address: %s\n", victimHost.IPAddr)
+		fmt.Printf("\t* Attacker IP address: %s\n", attackerHost.IPAddr)
+	})
+
+	prober := testutils.SynProber{
+		Iface:   attackerHost.HostIfaceName,
+		SrcMAC:  attackerHost.MACAddr,
+		DstMAC:  victimHost.MACAddr,
+		SrcIP:   attackerHost.IPAddr,
+		DstIP:   destIP,
+		SrcPort: 60000,
+		DstPort: destPort,
+	}
+
+	if err := attackerNs.InNetns(t, func() error {
+		// We need to manually warm-up victim's arp table, otherwise victim host might send an ARP request like:
+		// "Who has 192.168.210.3? Tell 127.0.0.1". This is obviously going to fail.
+		// warmUpPort is not the same as the port we want to test later on, as to make it easier to distinguish when
+		// reading tcpdump/wireshark output.
+		warmUpPort := prober.DstPort + 10
+		t.Logf("Warm-up victim's ARP table by trying to connect to %s:%d", victimHost.IPAddr, warmUpPort)
+
+		conn, err := net.Dial("tcp4", fmt.Sprintf("%s:%d", victimHost.IPAddr, warmUpPort))
+		if err != nil {
+			if !errors.Is(err, syscall.ECONNREFUSED) {
+				return fmt.Errorf("could not connect to %s:%d: %w", victimHost.IPAddr, warmUpPort, err)
+			}
+		} else {
+			conn.Close()
+			return fmt.Errorf("connection to %s:%d should fail, but did not. Test can't be conducted", victimHost.IPAddr, prober.DstPort)
+		}
+		t.Logf("Warm-up connection was correctly refused.")
+
+		err = prober.Probe()
+		if errors.Is(err, testutils.ErrNoSynAck) {
```

Comment:
I'm a bit concerned by the tests in this file as this error could be returned in two cases:

1. The SYN packet was received and ignored by the "victim" ;
2. The SYN packet wasn't received at all due to an issue with the test env. This is a potential source of false positive, which are pretty bad since we're testing security related behaviors.

I probably need to assert that the crafted packet is received by the victim to make sure we never hit 2.

Reply:
This is addressed by the last commit.